core/trino-main/src/main/java/io/trino/security/ViewAccessControl.java

```diff
@@ -72,6 +73,12 @@ public void checkCanExecuteFunction(SecurityContext context, String functionName
         wrapAccessDeniedException(() -> delegate.checkCanGrantExecuteFunctionPrivilege(context, functionName, invoker, false));
     }
 
+    @Override
+    public void checkCanExecuteFunction(SecurityContext context, FunctionKind functionKind, QualifiedObjectName functionName)
```

Comment:
(Me talking to myself) So the actual issue was that `ViewAccessControl` didn't override this method. It *was* actually called, but not intercepted for views...